Gemfile

```diff
@@ -26,6 +26,8 @@ gem 'puma', '~> 5.0'
 # Reduces boot times through caching; required in config/boot.rb
 gem 'bootsnap', '>= 1.4.4', require: false
 
+gem 'devise', '~> 4.8'
```

Comment:
I'm not sure if it's true, but probably Rafał Rothenberger [recommended devise_token_auth ](https://monterail.slack.com/archives/C02A0T92BMW/p1629885537019500) also 🤔

Reply:
> First of all, please merge last one PR and rebase this branch  there is a lot of not relevant changes

Great you mention it - after merge, should I rebase this branch locally and push with force (probably)? Because I'm not sure if it would be okay to push with force (I know, this is my project, but all in all it's a bad practice).

> I'm not sure if it's true, but probably Rafał Rothenberger recommended devise_token_auth
He did and I just watched again a fragment of his presentation - I misunderstood it - we do need tokens. 😬

Reply:
>Great you mention it - after merge, should I rebase this branch locally and push with force (probably)? Because I'm not sure if it would be okay to push with force (I know, this is my project, but all in all it's a bad practice), or maybe I just don't know some remote trick or flow.

Rebasing ==  push with force
Merge == push without force

I did not mentioned it, but you can do a few things - merge master to your branch or rebase with master. I prefer merging.

So it's better to:
1. Merge branch to master
2. `git checkout master` => go to master
3. `git pull` => fetch already merged things
4. `git checkout your-branch` => go to your current branch
5. `git merge master` => merge changes from master to your branch
6. Solve conflicts
7. `git push` => Push without force 🙏

Reply:
> 4. `git checkout your-branch` => go to your current branch
> 5. `git merge master` => merge changes from master to your branch
> 6. Solve conflicts
> 7. `git push` => Push without force

Worked like a charm, thank you 🌟